src/Operation/FindOneAndUpdate.php

```diff
@@ -103,6 +111,10 @@ public function __construct($databaseName, $collectionName, $filter, $update, ar
             'upsert' => false,
         ];
 
+        if (isset($options['arrayFilters']) && ! is_array($options['arrayFilters'])) {
+            throw InvalidArgumentException::invalidType('"arrayFilters" option', $options['arrayFilters'], 'array');
+        }
```

Comment:
The end of `FindOneAndUpdate::__construct()` passes its `$options` argument directly on to `FindAndModify`, so we really need to support the `arrayFilters` option in `FindAndModify`'s constructor. It should be documented in both classes.

Technically, that means that `FindOneAndReplace` and `FindOneAndDelete` can also receive the option, but it won't be documented for those and the server should raise an error anyway. We can also rely on [PHPLIB-124](https://jira.mongodb.org/browse/PHPLIB-124) catching that in the future.